strawberryfields/configuration.py

```diff
@@ -15,138 +15,229 @@
+    environment variables.
 
-BOOLEAN_KEYS = ("debug", "use_ssl")
+    .. note::
+
+        The configuration object (a nested dictionary) will be created based
+        on the following (order defines the importance, going from most
+        important to least important):
+
+        1. keyword arguments passed to ``load_config``
+        2. data contained in environmental variables (if any)
+        3. data contained in a configuration file (if exists)
+
+    Keyword arguments:
+        filename (str): the name of the configuration file to look for
+        authentication_token (str): the token to be used for user
+            authentication
+        hostname (str): the name of the host to connect to
+        use_ssl (bool): specifies if requests should be sent using SSL
+        port (int): the port to be used when connecting to the remote service
+        debug (bool): determines if the debugging mode is requested
+
+    Returns:
+        dict[str, dict[str, Union[str, bool, int]]]: the configuration
+            object
+    """
+    config = create_config_object()
+
+    parsed_config, _ = look_for_config_in_file(filename=filename)
+
+    if parsed_config is not None:
+        update_with_other_config(config, other_config=parsed_config)
+    else:
+        log.info("No Strawberry Fields configuration file found.")
 
+    update_from_environment_variables(config)
+
+    config_from_keyword_arguments = {"api": kwargs}
+    update_with_other_config(config, other_config=config_from_keyword_arguments)
+
+    return config
+
+def create_config_object(authentication_token="", **kwargs):
+    """Create a configuration object that stores configuration related data
+    organized into sections.
+
+    The configuration object contains API-related configuration options. This
+    function takes into consideration only pre-defined options.
+
+    If called without passing any keyword arguments, then a default
+    configuration object is created.
+
+    Keyword arguments:
+        authentication_token (str): the token to be used for user
+            authentication
+        hostname (str): the name of the host to connect to
+        use_ssl (bool): specifies if requests should be sent using SSL
+        port (int): the port to be used when connecting to the remote service
+        debug (bool): determines if the debugging mode is requested
+
+    Returns:
+        dict[str, dict[str, Union[str, bool, int]]]: the configuration
+            object
+    """
+    hostname = kwargs.get("hostname", "localhost")
+    use_ssl = kwargs.get("use_ssl", True)
+    port = kwargs.get("port", 443)
+    debug = kwargs.get("debug", False)
+
+    config = {
+        "api": {
+            "authentication_token": authentication_token,
+            "hostname": hostname,
+            "use_ssl": use_ssl,
+            "port": port,
+            "debug": debug
+            }
+    }
+    return config
+
+def look_for_config_in_file(filename="config.toml"):
```

Comment:
I might suggest an updated method name here, to signify that this function doesn't just look for the config file, but actually loads and returns it. Something concise :)